src/core/config/wrapping_type.rs

```diff
@@ -0,0 +1,147 @@
+use std::fmt::Formatter;
+use std::ops::Deref;
+
+use async_graphql::parser::types as async_graphql_types;
+use async_graphql::Name;
+use serde::{Deserialize, Serialize};
+
+use crate::core::is_default;
+
+/// Type to represent GraphQL type usage with modifiers
+/// [spec](https://spec.graphql.org/October2021/#sec-Wrapping-Types)
+#[derive(Clone, Serialize, Deserialize, PartialEq, Eq, schemars::JsonSchema)]
+#[serde(untagged)]
+pub enum WrappingType {
```

Comment:
How about we rename this to `Type` and `config::Type` to `config::TypeDef`

Reply:
renamed this type but left the `config::Type` as is since it'll be even more changes across codebase